dbfit-java/core/src/main/java/dbfit/api/PreparedDbCommand.java

```diff
@@ -0,0 +1,18 @@
+package dbfit.api;
+
+import java.sql.SQLException;
+
+/**
+ * Represents a prepared database statement with ability to bind input parameters and
+ * to register and access output ones including return values.
+ */
+public interface PreparedDbCommand extends DbCommand, DbQuery {
```

Comment:
> Would it make any sense to try to segregate PreparedStatement further? E.g. slice off getGeneratedKey() as we're only using it for INSERTs.
> 
> I've not really thought it through yet. Do you think it would make the statement creation much more complex?

One issue is when we create a statement and we want to ask it for both `execute()` and `getGeneratedKey()`.